kinematics/kinematics_solvers.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np
+from kinematics.kinematics_utils import homogeneous_transform_matrix, rot_z, rot_x
+from kinematics.kinematics_exceptions import (InvalidInitVectorElements, InvalidInitVectorShape, InvalidInitVectorLength,
+                                              InvalidInitVector)
+
+
+class KinematicsSolver:
+    """
+    Class storing parameters of the
+    robotic leg of Hexapod Robot.
+    The leg has 3 degrees of freedom.
+    """
+
+    def __init__(self, m1: np.ndarray, a1: np.ndarray, a2: np.ndarray, a3: np.ndarray):
+        """
+        @param m1: Translation from world origin to J1
+        @param a1: Link 1 (between J1 and J2)
+        @param a2: Link 2 (between J2 and J3)
+        @param a3: Link 3 (between J3 and FCP)
+        """
+
+        self.m1 = m1
+        self.a1 = a1
+        self.a2 = a2
+        self.a3 = a3
+
+    @staticmethod
+    def _check_vector_dimensions(vector: np.ndarray) -> None:
+        if not isinstance(vector, np.ndarray):
+            raise InvalidInitVector("Improper init vector type! Should be equal to np.ndarray!")
+        if vector.ndim != 1:
+            raise InvalidInitVectorShape("Improper vector shape! Ndim should be equal to 1!")
+        if len(vector) != 3:
+            raise InvalidInitVectorLength("Improper vector length! Vector length should equal to 3!")
+
+    @staticmethod
+    def _check_vector_numbers(vector: np.ndarray, scalar_index: int) -> None:
+        vector_indexes = [x for x in range(len(vector)) if x != scalar_index]
+        for index in vector_indexes:
+            if vector[index] != 0:
+                raise InvalidInitVectorElements(f"Invalid scalar's value! Scalar at index {index} should be equal to 0!")
```

Comment:
Mam wrażenie, jakby dało się to lepiej zrobić, ale nie jestem pewien czy trzeba

Reply:
No można zrobić to też w ten sposób:
```python
@staticmethod
def _check_vector_numbers(vector: np.ndarray, scalar_index: int) -> None:
    for index in range(len(vector)):
        if vector[index] != 0 and index != scalar_index:
            raise InvalidInitVectorElements(f"Invalid scalar's value! Scalar at index {index} should be equal to 0!")

```

Reply:
też ujdzie

Reply:
To masz jakieś inne rozwiązanie? Musisz sprawdzić, że każdy element wektora, oprócz wskazanego indeksu jest równy 0. Przy czym nie ograniczałbym się do wektora o długości 3.

Reply:
Znaczy to jest python, więc troche nie ma znaczenia jak. Ważne by działało.